src/ComposerRequireChecker/Cli/CheckCommand.php

```diff
@@ -260,12 +263,21 @@ private function getComposerData(string $jsonFile): array
         return JsonLoader::getData($jsonFile);
     }
 
-    private function getASTFromFilesLocator(InputInterface $input): LocateASTFromFiles
+    private function getASTFromFilesLocator(InputInterface $input, Options $options): LocateASTFromFiles
     {
         $errorHandler = $input->getOption('ignore-parse-errors') ? new CollectingErrorHandler() : null;
         $parser       = (new ParserFactory())->create(ParserFactory::PREFER_PHP7, new Lexer());
 
-        return new LocateASTFromFiles($parser, $errorHandler);
+        return new LocateASTFromFiles($parser, $errorHandler, $this->getCache($options));
+    }
+
+    private function getCache(Options $options): ?CacheInterface
+    {
+        if ($options->getCacheDirectory() === null) {
+            return null;
+        }
+
+        return new Psr16Cache(new FilesystemAdapter('composer-require-checker', 7 * 86400, $options->getCacheDirectory()));
```

Comment:
The `7 * 86400` is a bit arbitrary: would say "forever" instead?

Reply:
The default is `0` (forever) but that's not ideal because it will mean that caches will never be cleaned up. When running this in a CI pipeline you'll end up with a very large cache. Therefore, we need to find a value that's not too short and not forever.

Reply:
`86400 * 7` is basically the same thing as "forever", except that it leads to weekly hiccups :-\

Reply:
Changed to `forever`